frontend/javascripts/oxalis/view/components/command_palette.tsx

```diff
@@ -0,0 +1,198 @@
+    const commands: Command[] = [];
+
+    (Object.keys(userConfig) as [keyof UserConfiguration]).forEach((key, counter) => {
+      if (typeof userConfig[key] === "boolean") {
+        commands.push({
+          id: counter,
+          name: `Toggle ${getPhraseFromCamelCaseString(key)}`,
+          command: () => Store.dispatch(updateUserSettingAction(key, !userConfig[key])),
+          color: "#5660ff",
+        });
+      }
+    });
+    return commands;
+  };
+
+  const getNavigationEntries = () => {
+    if (activeUser == null) return [];
+    const commands: Command[] = [];
+    const basicNavigationEntries = [
+      { name: "Tasks (Dashboard)", path: "/dashboard/tasks" },
+      { name: "Annotations", path: "/dashboard/annotations" },
+      { name: "Datasets", path: "/dashboard/datasets" },
+      { name: "Time Tracking", path: "/timetracking" },
+    ];
+
+    const adminMenu = getAdministrationSubMenu(false, activeUser);
+    const adminCommands: Array<{ name: string; path: string }> = [];
+    adminMenu?.children.map((entry: { key: string }) => {
+      adminCommands.push({ name: getLabelForPath(entry.key), path: entry.key });
+    });
+
+    const statisticsCommands = Utils.isUserAdminOrManager(activeUser)
+      ? [
+          {
+            path: "/reports/projectProgress",
+            name: "Project Progress",
+          },
+          {
+            path: "/reports/availableTasks",
+            name: "Available Tasks",
+          },
+        ]
+      : [];
+
+    const navigationEntries = [...basicNavigationEntries, ...adminCommands, ...statisticsCommands];
+
+    navigationEntries.forEach((entry, counter) => {
+      commands.push({
+        id: counter,
+        name: `Navigate to ${entry.name}`,
+        command: () => {
+          window.location.href = entry.path;
+        },
+        color: "#5660ff",
+      });
+    });
+
+    return commands;
+  };
+
+  const getToolEntries = () => {
+    if (!isInTracingView) return [];
+    const commands: Command[] = [];
+    let availableTools = Object.keys(AnnotationToolEnum) as [keyof typeof AnnotationToolEnum];
+    if (isViewMode || !restrictions.allowUpdate) {
+      availableTools = AvailableToolsInViewMode as [keyof typeof AnnotationToolEnum];
+    }
+    availableTools.forEach((tool, counter) => {
+      commands.push({
+        id: counter,
+        name: `Switch to ${getLabelForTool(tool)} Tool`,
+        command: () => Store.dispatch(setToolAction(tool)),
+        color: "#5660ff",
+      });
+    });
+    return commands;
+  };
+
+  const menuActions = getMenuActions(isViewMode);
+
+  const allCommands = [
+    ...getNavigationEntries(),
+    ...getToolEntries(),
+    ...mapMenuActionsToCommands(menuActions),
+    ...getTabsAndSettingsMenuItems(),
+  ];
+  return (
+    <div style={{ marginRight: "10px" }}>
+      <ReactCommandPalette
+        commands={allCommands.map((command, counter) => {
+          return {
+            ...command,
+            id: counter,
```

Comment:
so, this overwrites all previously defined ids (which is good because they would be not unique otherwise). can we avoid defining the other ids in the first place?